packages/server/src/insights/dashboard.entity.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  BaseEntity,
+  Column,
+  Entity,
+  ManyToOne,
+  PrimaryColumn,
+  Unique,
+} from 'typeorm';
+import { UserCourseModel } from '../profile/user-course.entity';
+import { InsightDetail } from '@koh/common';
+
+@Entity('insight_dashboard_model')
+@Unique('dashboardKey', ['name', 'userCourseId'])
```

Comment:
Neato! I was unaware of the `@Unique` decorator

Reply:
it creates the same constraint as the PK... I think this is when I was considering whether or not to use an ID rather than a name as PK; it's probably not necessary but doesn't hurt either way